extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java

```diff
@@ -168,13 +170,22 @@ private String guessDriver(Optional<JdbcDataSourceBuildItem> jdbcDataSource) {
     }
 
     @BuildStep
-    @Record(RUNTIME_INIT)
-    public void build(QuartzRuntimeConfig runtimeConfig, QuartzBuildTimeConfig buildTimeConfig, QuartzRecorder recorder,
-            BeanContainerBuildItem beanContainer,
-            BuildProducer<ServiceStartBuildItem> serviceStart, QuartzJDBCDriverDialectBuildItem driverDialect,
-            List<JdbcDataSourceSchemaReadyBuildItem> schemaReadyBuildItem) {
-        recorder.initialize(runtimeConfig, buildTimeConfig, beanContainer.getValue(), driverDialect.getDriver());
+    public void start(BuildProducer<ServiceStartBuildItem> serviceStart,
+            @SuppressWarnings("unused") List<JdbcDataSourceSchemaReadyBuildItem> schemaReadyBuildItem) {
         // Make sure that StartupEvent is fired after the init
         serviceStart.produce(new ServiceStartBuildItem("quartz"));
     }
+
+    @BuildStep
+    @Record(RUNTIME_INIT)
+    public void quartzSupportBean(QuartzRuntimeConfig runtimeConfig, QuartzBuildTimeConfig buildTimeConfig,
+            QuartzRecorder recorder,
+            BuildProducer<SyntheticBeanBuildItem> syntheticBeanBuildItemBuildProducer,
+            QuartzJDBCDriverDialectBuildItem driverDialect) {
+
+        syntheticBeanBuildItemBuildProducer.produce(SyntheticBeanBuildItem.configure(QuartzSupport.class)
```

Comment:
Ah did not know we have this. Thanks :)

Reply:
Yeah. @mkouba has improved the experience of "creating" beans at "build time".